File: algorithms/leetcode/maximum_points_cards_1423.go
package main

import "fmt"




func maxScore(cardPoints []int, k int) int {
  max := 0 
  for i := 0; i < k; i +=1 {
      max += cardPoints[i] 
  }
  curr := max
  for i := 0; i < k; i +=1 {
   curr -= cardPoints[k-i -1] 
   curr += cardPoints[len(cardPoints)-i -1]
   if curr > max {
     max = curr
   }

  }
  return max
}



func main() {

  fmt.Println(maxScore([]int{1,2,3,4,5,6,1}, 3))

}

File: algorithms/leetcode/maximum_points_cards_1423.py




def maxScore(cardPoints, K):
    curr = best = sum(cardPoints[:K])
    k = K -1
    step = 1 
    while k >= 0:
       curr -= cardPoints[k] 
       k -=1
       curr += cardPoints[-step]
       step +=1
       best = max(best, curr)
    return best




cardPoints = [2,2,2]
k = 2 
res1 = maxScore(cardPoints, k)
print(res1)

k = 0 
curr = 8 

    
